server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/report/RankedBallotInterpretationReportTests.java

```diff
@@ -0,0 +1,166 @@
+/*
+Democracy Developers IRV extensions to colorado-rla.
+
+@copyright 2024 Colorado Department of State
+
+These IRV extensions are designed to connect to a running instance of the raire
+service (https://github.com/DemocracyDevelopers/raire-service), in order to
+generate assertions that can be audited using colorado-rla.
+
+The colorado-rla IRV extensions are free software: you can redistribute it and/or modify it under the terms
+of the GNU Affero General Public License as published by the Free Software Foundation, either
+version 3 of the License, or (at your option) any later version.
+
+The colorado-rla IRV extensions are distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along with
+raire-service. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package au.org.democracydevelopers.corla.report;
+
+import au.org.democracydevelopers.corla.util.TestClassWithDatabase;
+import au.org.democracydevelopers.corla.util.testUtils;
+
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.ext.ScriptUtils;
+import org.testcontainers.jdbc.JdbcDatabaseDelegate;
+import org.testcontainers.shaded.org.apache.commons.lang3.StringUtils;
+import us.freeandfair.corla.csv.DominionCVRExportParser;
+import us.freeandfair.corla.persistence.Persistence;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+import org.testng.annotations.*;
+import us.freeandfair.corla.query.ExportQueries;
+
+import javax.transaction.Transactional;
+
+import static au.org.democracydevelopers.corla.util.testUtils.*;
+import static org.testng.Assert.*;
+
+import static us.freeandfair.corla.query.CountyQueries.fromString;
+
+/**
+ * Test that a correct ranked vote interpretation report is produced, for a variety of IRV-related
+ * parsing examples, including:
+ * ThreeCandidatesTenVotes - a constructed test file will all-valid IRV votes.
+ * ThreeCandidatesTenInvalidVotes - a constructed test with invalid IRV ballots to check they are accepted and properly
+ *   interpreted.
+ * An equivalent test using real data from Boulder '23 is in DominionCVRExportParserTests.
+ * Test csvs are a subset of those in DominionCVRExportParserTests.
+ */
+public class RankedBallotInterpretationReportTests extends TestClassWithDatabase {
+
+  /**
+   * Class-wide logger
+   */
```

Comment:
Generally we don't have the au.org. etc prefix for the Class.class statement here.